rcl/CMakeLists.txt

```diff
@@ -3,14 +3,68 @@ cmake_minimum_required(VERSION 2.8.3)
 project(rcl)
 
 find_package(ament_cmake REQUIRED)
+find_package(rcl_interfaces REQUIRED)
+find_package(rmw REQUIRED)
+find_package(rmw_implementation REQUIRED)
+find_package(rmw_implementation_cmake REQUIRED)
+find_package(rosidl_generator_c REQUIRED)
 
+include_directories(include)
+
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
+endif()
+
+set(${PROJECT_NAME}_sources
+  src/rcl/allocator.c
+  src/rcl/common.c
+  src/rcl/guard_condition.c
+  src/rcl/node.c
+  src/rcl/publisher.c
+  src/rcl/rcl.c
+  src/rcl/subscription.c
+  src/rcl/wait.c
+  src/rcl/time.c
+  src/rcl/timer.c
+)
+
+macro(target)
+  add_library(${PROJECT_NAME}${target_suffix} SHARED ${${PROJECT_NAME}_sources})
+  ament_target_dependencies(${PROJECT_NAME}${target_suffix}
+    "rcl_interfaces"
+    "rmw"
+    "rosidl_generator_c"
+    "${rmw_implementation}"
+  )
+
+  # Causes the visibility macros to use dllexport rather than dllimport,
+  # which is appropriate when building the dll but not consuming it.
+  target_compile_definitions(${PROJECT_NAME}${target_suffix} PRIVATE "RCL_BUILDING_LIBRARY")
```

Comment:
It was actually broken before: 2398e98 I must have copied the CMake code from `rclcpp` when refactoring for the typesupport changes (refactored to build one library for each impl like `rclcpp`) after testing on Windows, because in the current state it would not work.